docs/src/examples/components/Icon/Usage/IconButtonExample.shorthand.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { Icon, Button, Text } from '@stardust-ui/react'
+
+const IconExampleButton = () => (
+  <div>
+    <Button type="primary" icon iconPosition="before">
```

Comment:
actually, this is not a _shorthand_ example.

Reply:
I agree button with icon should be part of Button docs. My goal with this is to show how icons can be used and inherit color (via currentColor) from their context (body text or containing component). Do you have another suggestion on what ways I could show that? Or do you not believe it is necessary?

Reply:
let me, please, try to quickly address it on my local env - will back right after

Reply:
let me, please, bring the results back. 

First thing to note is that it would be possible to achieve the same goal this example provides by using the following code:

```jsx
const IconExampleButton = () => (
  <div>
    <Button type="primary" icon={{ name: "call", size: 'large' }} content="Call now" />
    <Button type="secondary" icon="home" content="Phone home" />
    <Button type="primary" disabled icon="home" content="Disabled button with icon" />
  </div>
)
```

Most critical thing to note is that _if there are children being explicitly provided to the component, then its props won't affect anyhow the rendered tree of this component_ - `children` are always rendered 'as is'. Speaking of the `Button` example we originally had

```jsx
 <Button type="secondary" icon iconPosition="before">
      <Icon name="home" xSpacing="after" />
      <Text content="Phone home" />
    </Button>
``` 

Please, note that as a consequence of this principle is that both `icon` and `iconPosition` _are simply ignored and make any effect to the rendered tree of a `Button`_, as long as there are children (`<Icon>` and `<Text>`).

------

## Maybe a better example

Please, consider the following example that, probably, better reflects the idea you would like to present:

```jsx
<>
    <Button variables={{ color: 'red' }} icon='home' content='Phone home' />
    <Button variables={{ color: 'orange' }} icon='home' content='Phone home' />
    <Button variables={{ color: 'green' }} icon='home' content='Phone home' />
    <Button variables={{ color: 'lightblue' }} icon='home' content='Phone home' />
</>
```
If this one works, we could consider to do the following changes
- rename an example to something that better reflects the idea, like ("Icon and context-dependent color"), better description like "Icons can be used and inherit color from their context (body text or containing component)." (note, this is word-for-word your phrase:)

Please, let me know about your thoughts.

Reply:
Thanks for your detailed feedback. Just talked with @levithomason and have decided to pull this example out of this component and do a separate PR for the Button component for different colors and to have some of those buttons show using an icon.